components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java

```diff
@@ -0,0 +1,1323 @@
+    }
+
+    @DataProvider(name = "dataProviderForGetRoleUnexpectedServerError")
+    public Object[][] dataProviderForGetRoleUnexpectedServerError() {
+
+        return new Object[][]{
+                {SAMPLE_VALID_ROLE_ID2, SAMPLE_TENANT_DOMAIN2, "sql error"},
+                {SAMPLE_VALID_ROLE_ID, SAMPLE_INVALID_TENANT_DOMAIN, null},
+                {SAMPLE_VALID_ROLE_ID2, SAMPLE_INVALID_TENANT_DOMAIN, "sql error"},
+        };
+    }
+
+    @Test(dataProvider = "dataProviderForGetRoleUnexpectedServerError")
+    public void testGetRoleUnexpectedServerError(String roleId, String tenantDomain, String sError)
+            throws IdentityRoleManagementException {
+
+        when(mockRoleManagementService.getRole(roleId, tenantDomain)).
+                thenThrow(unExpectedErrorThrower(tenantDomain, sError,
+                        "Error while creating the role: %s in the tenantDomain: %s", roleId));
+
+        SCIMRoleManager scimRoleManager = new SCIMRoleManager(mockRoleManagementService, tenantDomain);
+        assertThrows(CharonException.class, () -> scimRoleManager.getRole(roleId, null));
+    }
+
+    @DataProvider(name = "dataProviderForGetRolePositive")
+    public Object[][] dataProviderForGetRolePositive() {
+
+        return new Object[][]{
+                {SAMPLE_VALID_ROLE_ID, SAMPLE_VALID_ROLE_NAME, "roleDomain1", SAMPLE_TENANT_DOMAIN,
+                        "urn:ietf:params:scim:schemas:extension:2.0:Role:groups.value", true, false},
+                {SAMPLE_VALID_ROLE_ID2, SAMPLE_VALID_ROLE_NAME2, null, SAMPLE_TENANT_DOMAIN,
+                        "urn:ietf:params:scim:schemas:extension:2.0:Role:groups.value", false, false},
+                {SAMPLE_VALID_ROLE_ID2, SAMPLE_SYSTEM_ROLE_NAME, null, SAMPLE_TENANT_DOMAIN,
+                        "urn:ietf:params:scim:schemas:extension:2.0:Role:groups.value", false, true},
+                {SAMPLE_VALID_ROLE_ID, null, "roleDomain1", SAMPLE_TENANT_DOMAIN2, null, false, true},
+                {SAMPLE_VALID_ROLE_ID2, "", "roleDomainX", SAMPLE_TENANT_DOMAIN, "", true, false},
+                {null, SAMPLE_VALID_ROLE_NAME, "", SAMPLE_TENANT_DOMAIN2, null, true, false},
+                {"", "", "", "", "", false, true}
+        };
+    }
+
+    @Test(dataProvider = "dataProviderForGetRolePositive")
+    public void testGetRolePositive(String roleId, String roleName, String domain, String tenantDomain,
+                                    String attributeKey, Boolean attributeValue, boolean isEmptyLists)
+            throws IdentityRoleManagementException, BadRequestException, NotFoundException, CharonException {
+
+        org.wso2.carbon.identity.role.mgt.core.Role role = getDummyIdentityRole(roleId, roleName, domain, tenantDomain,
+                isEmptyLists);
+        Map<String, Boolean> attributeMap = null;
+        if (attributeKey != null) {
+            attributeMap = new HashMap<>();
+            attributeMap.put(attributeKey, attributeValue); //dummy details
```

Comment:
Formatting error. Need to start from an upper case letter, there needs to be a space after `//` and need to end with a `.`. Better if we can add the comment in a new line.

Reply:
Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/387